ui/main/src/app/services/lightcards.service.ts

```diff
@@ -0,0 +1,185 @@
+        this.computeFilteredLightCards();
+    }
+
+    private computeFilteredAndSortedLightCards() {
+        combineLatest([
+            this.store.pipe(select(feedSelectors.selectSortFilter)),
+            this.getFilteredLightCards()
+        ]
+        ).pipe(
+            map(results => {
+                function compareFn(needToSortBySeverity: boolean, needToSortByRead: boolean) {
+                    if (needToSortByRead) {
+                        if (needToSortBySeverity) {
+                            return compareByReadSeverityPublishDate;
+                        } else {
+                            return compareByReadPublishDate;
+                        }
+                    } else if (needToSortBySeverity) {
+                        return compareBySeverityPublishDate;
+                    }
+                    return compareByPublishDate;
+                }
+                return results[1]
+                    .sort(compareFn(results[0].sortBySeverity, results[0].sortByRead));
+            }
+            )
+        ).subscribe((lightCards) => this.filteredAndSortedLightCards.next(lightCards));
+    }
+    public getFilteredLightCards(): Observable<any> {
+        return this.filteredLightCards.asObservable();
+    }
+
+    private computeFilteredLightCards() {
+        combineLatest([
+            this.store.pipe(select(feedSelectors.selectActiveFiltersArray)),
+            this.getLightCards()
+        ]
+        ).pipe(
+            map(results => {
+                console.log(new Date().toISOString(), "Number of card in memory ", results[1].length, " cards");
+                return this.filterLightCards(results[1], results[0]);
+            })
+        ).subscribe((lightCards) => this.filteredLightCards.next(lightCards));
+    }
+
+ // --------------------
+ // When an flow of card is coming, for performance reasons , we do not want to update the card list 
+ // every time  a card is arriving so we wait for the end of the flow of cards.
+ // But if it takes too long, we want to show something so every seconds we make a rendering 
+ // even if the flow is still continuing.
+ // Do do that we combine a debounce waiting for the end of the flow and an interval to get the card list every seconds 
+
+    private getLightCards(): Observable<any> {
+        return merge(this.getLightCardsInterval(), this.getLightCardDebounce());
+    }
+
+    private getLightCardsInterval(): Observable<any> {
+        return this.store.pipe(
+            select(feedSelectors.selectFeed),
+            sample(interval(1000)),
+            filter(() => ((new Date().valueOf()) - this.lastDebounce) > 1000) // we only need to get cards if no debounce arise in 1 seconds) 
+        );
+    }
+
+    private getLightCardDebounce(): Observable<any> {
+        return this.store.pipe(
+            select(feedSelectors.selectFeed),
+            debounceTime(200),
+            tap(() => this.lastDebounce = (new Date()).valueOf())
+        );
+    }
```

Comment:
Wouldn't the [throttleTime](https://rxjs.dev/api/operators/throttleTime) operator with 200 as parameter do the trick?
It would let lightcard updates pass as long as there are less than 1 every 200ms, then take one every 200ms if there are more.
In the end, you emit at most at a rate of 1 every 200ms.

Reply:
But after all it works so... if it ain't broke don't fix it ;-)

Reply:
it will not have the behavior i want : 

- wait for the end of flow (debounce) 
- if flow is  too long, update at least every 1000ms  (sample(interval))
- if a few card arrives , do not wait 1000ms , process  after 200ms (debounce) 

using throttleTime(200) will process the flux every 200s instead of every 1000ms if there is a lot of card 
using throttleTime(1000) will wait 1000ms even if we have one card to process